engine/engine.go

```diff
@@ -97,6 +109,44 @@ func (e *Engine) Run(ctx context.Context, step *spec.Step, output io.Writer) (*r
 	return exec.Run(ctx, step, output)
 }
 
+func fileFolderCreation(paths []*spec.File) error {
+	for _, f := range paths {
+		if f.Path != "" {
+			path := f.Path
+			if _, err := os.Stat(path); err == nil {
+				continue
+			}
+			if f.IsDir {
+				// create a folder
+				if err := os.MkdirAll(path, fs.FileMode(f.Mode)); err != nil {
+					return errors.Wrap(err,
+						fmt.Sprintf("failed to create directory for host path: %s", path))
+				}
+			} else {
+				// create a file
+				newFile, newErr := os.Create(path)
```

Comment:
Use os.OpenFile and create a file with correct permissions in a single call https://pkg.go.dev/os#OpenFile

Reply:
If the file is supposed to be created as read-only, then we create it with `OpenFile` and the correct permissions (and, of course, drop the call to `Chmod` below), will write to the file fail?

I'm thinking to modify the code. To create the file, write to it, close it, and then call `Chmod` to set the permissions.